### PR TITLE
Add GRPC Stdio Copying system

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

**Comment by jefferai** [comment]
A note to any reviewers, some of the comments indicate that this isn't necessarily fully implemented.

File: grpc_client.go [hunk 1]
```diff
@@ -67,6 +74,32 @@ func newGRPCClient(doneCtx context.Context, c *Client) (*GRPCClient, error) {
...
+		response, e := stream.Recv()
+		switch e {
+		case nil:
+		case io.EOF, context.Canceled:
+			return
+		default:
+			logger.Error("Experienced issue in gRPC copier", "error", e)
+			return
+		}
+		writer.Write(response.Data)
```
Comment (yurishkuro) [inline grpc_client.go]: this would be better placed inside `case nil:`

**Comment by yurishkuro** [comment]
I wonder, wouldn't it be easier for the client to simply read the stdout/err of the child process, split on `\n` and forward to its own stdout/err? This way the plugins won't need to implement anything special, especially when they are written in other languages.

**Comment by mitchellh** [comment]
Superseded by #135. Thank you!

**Pull request closed without merging**